### DISCUSSION TITLE
Using Redis for a Custom Script as key value store

### DISCUSSION THREAD

**Discussion by julianstolp** [netbox-community/netbox#14334] [Q&A] [answered]
Hello I'm testing to use the netbox redis database as replacement for a key value textfile. It works as stated below, but my only concern is that it might mess up internal processes in NetBox.
Should I just use another subsection in configuration.py eg. 'macip' and use db=15?

```
    'macip': {
        'HOST': 'localhost',
        'PORT': 6379,
        'USERNAME': ''
        'PASSWORD': '',
        'DATABASE': 15,
        'SSL': False,
    }
} 
....
redis_client = redis.Redis(host='localhost', port=6379, db=15)
```

```
import os
import redis
from mac_to_ip.tasks import task_mac_to_ip
from dotenv import load_dotenv

redis_client = redis.Redis(host='localhost', port=6379)

if __name__ == "__main__":
    load_dotenv('/opt/netbox/netbox/scan/.env')
    full_map={}
    simulator = task_mac_to_ip(os.environ.get('SIMULATOR_IP'),manufacturer="SIMU")
    full_map.update(simulator)
    for mac, ip in full_map.items():
        redis_client.set(mac, ip)
```

**Comment by candlerb** [comment, marked as answer]
As long as you're talking to a different Redis database than Netbox is using (and as you've observed, the default config uses db=0 and db=1 on localhost) then they're not going to conflict.

It would probably work to create a new section in the REDIS global setting in configuration.py, but technically that's part of Netbox's own configuration. If you're writing a plugin, then [PLUGINS_CONFIG](https://docs.netbox.dev/en/stable/plugins/#configure-plugin) would be the official place to put it.

If it's only a Custom Script then there's no official place to put shared configuration settings that I know of, so do whatever works for you (even within the script itself).

**Reply by julianstolp** [reply]
Thank you 👍🏼 
I will use the NetBox Redis Database with id=15
